torchvision/transforms/functional_tensor.py

```diff
@@ -993,3 +993,52 @@ def equalize(img: Tensor) -> Tensor:
         return _equalize_single_image(img)
 
     return torch.stack([_equalize_single_image(x) for x in img])
+
+
+def elastic_transform(
+    img: Tensor,
+    alpha: Optional[List[float]] = None,
+    sigma: Optional[List[float]] = None,
+    interpolation: str = "bilinear",
+    fill: Optional[List[float]] = None,
+    displacement: Optional[Tensor] = None,
+    random_state: Optional[int] = None,
+) -> Tensor:
+
+    if not (isinstance(img, torch.Tensor)):
+        raise TypeError(f"img should be Tensor. Got {type(img)}")
+
+    size = list(img.shape[-2:])
+    if random_state is not None:
+        torch.manual_seed(random_state)
+
+    if displacement is not None:
+        displacement = displacement.to(img.device)
+    elif alpha is not None and sigma is not None:
+        dx = torch.rand([1, 1] + size, device=img.device) * 2 - 1
+        if sigma[0] > 0.0:
+            dx = gaussian_blur(
+                dx,
+                [4 * int(sigma[0]) + 1, 4 * int(sigma[1]) + 1],
```

Comment:
@JLrumberger do you have any refs for this choice of kernel size ?
If we take https://gist.github.com/ernestum/601cdf56d2b424757de5#file-elastic_transform-py-L16-L17 as ref, then
it would be something like `~int(8 * sigma + 1)`:
```
sd = float(sigma)
lw = int(truncate * sd + 0.5)
x = numpy.arange(-radius, radius+1)  # radius = lw
```
- https://github.com/scipy/scipy/blob/47bb6febaa10658c72962b9615d5d5aa2513fa3a/scipy/ndimage/filters.py#L258

- https://github.com/scipy/scipy/blob/47bb6febaa10658c72962b9615d5d5aa2513fa3a/scipy/ndimage/filters.py#L187

Reply:
@vfdev-5 no, I don't have a reference for this, but I have a few comments:
1. I think ernestums implementation and ours are not really comparable when it comes to the parameters, since ernestum uses `scipy.ndimage.interpolation.map_coordinates`  which expects as an identity sampling grid this `np.meshgrid(np.arange(shape[0]), np.arange(shape[1]), np.arange(shape[2]))`, so values lie between 0 and H or W.  In our implementation we use `vision.torchvision.transforms.functional_tensor._apply_grid_transform` which expects as an identity grid this `[torch.linspace((-s + 1) / s, (s - 1) / s, s) for s in size]`, so values lie between -1:1 for any given image independant of the shape. Anyways, I think the smoothing shouldn't be affected by this
2. I looked into imgaugs Elastic Transformation and they calculate the kernel_size based on the sigma like this:
`def _compute_gaussian_blur_ksize(sigma):
    if sigma < 3.0:
        ksize = 3.3 * sigma  # 99% of weight
    elif sigma < 5.0:
        ksize = 2.9 * sigma  # 97% of weight
    else:
        ksize = 2.6 * sigma  # 95% of weight

    # we use 5x5 here as the minimum size as that simplifies
    # comparisons with gaussian_filter() in the tests
    # TODO reduce this to 3x3
    ksize = int(max(ksize, 5))
    return ksize` 
I don't think that this is necessary for our implementation.
3. I ran our implementation as is `elastic_transform(img,alpha=[40,40],sigma=[4,4], random_state=0)` and with the kernel size changed to `int(8 * sigma + 1)` and stored the results in the red channel of the as is implementation in the blue channel and the one with the bigger kernel size in the red channel and the results look identical:
![red_8_blue_4](https://user-images.githubusercontent.com/22618181/152697574-03a1b815-0a49-4be6-9bb1-36758fd99045.PNG)
 
 So all in all, I think we can change it to `int(8 * sigma + 1)` for consistency with ernestums implementation. Sorry for the late reply and thanks for all your work :)

Reply:
Thanks for updates @JLrumberger !
Major difference between `int(8 * sigma + 1)` vs `4 * int(sigma) + 1` is when using sigma < 1. For all sigmas < 1 we'll have kernel size as 1 in the second case. 
For sigma = 4, `int(8 * sigma + 1)=33` vs `4 * int(sigma) + 1 = 17`.